tools/integration_tests/operations/list_dir_test.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Provides integration tests for list directory.
+package operations_test
+
+import (
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+)
+
+func createDirectoryWithFile(dirPath string, filePath string, t *testing.T) {
+	err := os.Mkdir(dirPath, setup.FilePermission_0600)
+	if err != nil {
+		t.Errorf("Mkdir at %q: %v", dirPath, err)
+		return
+	}
+
+	_, err = os.Create(filePath)
+	if err != nil {
+		t.Errorf("Error in creating file %v:", err)
+	}
+}
+
+func createDirectoryStructureForTest(t *testing.T) {
+	// Directory structure
+	// testBucket/directoryForListTest                                                                     -- Dir
+	// testBucket/directoryForListTest/fileInDirectoryForListTest		                                       -- File
+	// testBucket/directoryForListTest/firstSubDirectoryForListTest                                        -- Dir
+	// testBucket/directoryForListTest/firstSubDirectoryForListTest/fileInFirstSubDirectoryForListTest     -- File
+	// testBucket/directoryForListTest/secondSubDirectoryForListTest                                       -- Dir
```

Comment:
do you think adding a test case for a sub directory with more than one file would also be helpful ?

Reply:
I think it is covered in the listing the main directory. so currently, how is it working
mntDir
    - dir1
         - subdir1
         - subdir2
         - emptysubdir
         - file
         
We are started listing from mntDir. So first, dir1 will get listed. and dir1 has all types of data.

Yet you think we should add, I will add for the same.

Reply:
Yes agree, mntDir has more than one file collectively in its child directories. But I was thinking there is not dir/subDir which immediately has more than one file.